gpytorch/models/model_list.py

```diff
@@ -91,8 +98,26 @@ def train_targets(self):
 
 
 def _get_tensor_args(*args):
+    arg_list = []
     for arg in args:
         if torch.is_tensor(arg):
-            yield (arg,)
+            arg_list.append((arg,))
         else:
-            yield arg
+            arg_list.append(arg)
+    return arg_list
+
+
+def _length_safe_zip(*args):
```

Comment:
Should this live in some shared utility? Like in `gpytorch/utils/generic.py` or the like?